### PR TITLE
Public List, Users and Role Detection, Record Statuses

### PR DESCRIPTION
**Pull request by thehabes**

### CODE REVIEW

**Comment by thehabes** [comment]
Tested and seems functional, using the same status templates as from deer-letters.

**Review by cubap — CHANGES_REQUESTED** [review]
small things, but worth changing

File: js/deer-config.js [hunk 1]
```diff
@@ -175,10 +178,10 @@ const config = {
                                         type: 'xml'
                                     })
                                     const poemText = SaxonJS.XPath.evaluate("/" + t.selector?.value, sampleSource, { xpathDefaultNamespace: 'http://www.tei-c.org/ns/1.0' })
-                                    textSample.innerHTML = poemText.innerHTML
+                                    elem.querySelector(".textSample").innerHTML = poemText.innerHTML
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-config.js:181]: I did not change anything here as this was required to get the page to load.  Let me know if we have to change something now or if we can just ticket this.

File: js/deer-config.js [hunk 2]
```diff
@@ -323,7 +326,7 @@ const config = {
         },
         managedlist: (obj, options = {}) => {
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-config.js:327]: Separate issue (and we need to do the same in Dunbar Letters, I will make an issue.)

File: js/deer-config.js [hunk 3]
```diff
@@ -323,7 +326,7 @@ const config = {
...
             // Come on, Mr. Hacker. We both know you could break in here, but why waste your time? It is tested on the server past here.
-            if(!userHasRole(["dunbar_user_admin","dunbar_user_contributor","dunbar_user_public"])) { return `This function is limited to administrators.`}
+            if(!userHasRole(["dunbar_user_admin","dunbar_user_contributor","dunbar_user_public"])) { return `This function is limited to contributors.`}
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-config.js:329]: separate issue

File: js/deer-render.js [hunk 4]
```diff
@@ -248,6 +248,217 @@ DEER.TEMPLATES.event = function (obj, options = {}) {
...
+                    body: JSON.stringify(coll)
+                })
+                .then(r => r.ok ? r.json() : Promise.reject(Error(r.text)))
+                .then(data => {
+                    el.innerHTML = `✔ submitted`
+                    el.classList.remove("button")
+                    el.classList.remove("bg-error")
+                    el.classList.add("bg-success")
+                    el.addEventListener("click", e => {
+                        return
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-render.js]: Changed, I think the goal was to remove the `click` listener.

File: js/deer-render.js [hunk 5]
```diff
@@ -248,6 +248,217 @@ DEER.TEMPLATES.event = function (obj, options = {}) {
...
+            .then(response => response.json())
+            .then(coll => {
+                //If you want to know who placed it, you have to check the author of the moderating Annotation
+                if(coll.itemListElement.filter(record => record["@id"] === obj['@id']).length){
+                    statusArea.innerHTML += `<div title="This record is available publicly" class="recordStatus tag is-small bg-success"> ✔ public </div>`
+                    return true
+                }
+                else{
+                    //statusArea.innerHTML += `<div title="This record is not available to the public" class="recordStatus tag is-small bg-error"> ❌ public </div>`
+                    return false
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-render.js]: else is removed, but the result should be boolean so false is still returned

File: js/deer-render.js [hunk 6]
```diff
@@ -248,6 +248,217 @@ DEER.TEMPLATES.event = function (obj, options = {}) {
...
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(reviewedQuery)
+            })
+            .then(response => response.json())
+            .then(data => {
+                if(data.length){
+                    elem.setAttribute(DEER.SOURCE, data[0]['@id'])
+                    rejected = (!data[0].body.releasedTo && data[0].body.resultComment) ? true : false
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-render.js]: see slack comments.  Does not need ternary, but result should be boolean.

File: js/deer-render.js [hunk 7]
```diff
@@ -248,6 +248,217 @@ DEER.TEMPLATES.event = function (obj, options = {}) {
...
+                    if(data[0].body.releasedTo ? true : false){
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-render.js]: cleaned up.

File: js/deer-render.js [hunk 8]
```diff
@@ -248,6 +248,217 @@ DEER.TEMPLATES.event = function (obj, options = {}) {
...
+                        statusArea.innerHTML += `<div class="recordStatus tag is-small bg-success" title="The submission was accepted">✔ accepted</div>`
+                    }
+                    //reviewed = accepted || rejected
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-render.js]: cleaned up

**Pull request merged**